src/main/java/org/javarosa/core/model/instance/GeoJsonExternalInstance.java

```diff
@@ -0,0 +1,59 @@
+package org.javarosa.core.model.instance;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.Map;
+import org.javarosa.core.model.data.UncastData;
+import org.javarosa.core.util.GeoUtils;
+
+public class GeoJsonExternalInstance {
+    public static TreeElement parse(String instanceId, String path) throws IOException {
+        return parse(instanceId, new FileInputStream(path));
+    }
+
+    protected static TreeElement parse(String instanceId, InputStream geojsonStream) throws IOException {
+        final TreeElement root = new TreeElement("root", 0);
+        root.setInstanceName(instanceId);
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        JsonNode geoJsonTree = objectMapper.readTree(geojsonStream);
+
+        if (!geoJsonTree.get("type").asText().equals("FeatureCollection")) {
+            throw new IOException("GeoJSON file must contain a top-level FeatureCollection");
+        }
+
+        JsonNode features = geoJsonTree.get("features");
+        int multiplicity = 0;
+
+        for (JsonNode feature : features) {
+            root.addChild(toTreeElement(feature, multiplicity));
+            multiplicity++;
+        }
+
+        return root;
+    }
+
+    private static TreeElement toTreeElement(JsonNode feature, int multiplicity) throws IOException {
+        TreeElement item = new TreeElement("item", multiplicity);
+
+        String geometry = GeoUtils.geojsonToOdkGeometry(feature.get("geometry"));
+        TreeElement geoField = new TreeElement("geometry", 0);
+        geoField.setValue(new UncastData(geometry));
+        item.addChild(geoField);
+
+        JsonNode properties = feature.get("properties");
+
+        for (Iterator<Map.Entry<String, JsonNode>> i = properties.fields(); i.hasNext();) {
```

Comment:
`forEachRemaining` fails our Android lint check. I marginally prefer this to a `while`.